threes.py

```diff
@@ -68,25 +73,25 @@ def play_game():
     import random
 
     deck = make_deck()
-    pos = random.sample(xrange(16), 9)
+    pos = random.sample(range(16), 9)
     m = np.zeros((16,), dtype=int)
     m[pos] = deck[:len(pos)]
     deck = deck[len(pos):]
     m = m.reshape((4,4))
 
     while True:
-        lineset = [get_lines(m, i) for i in xrange(4)]
-        foldset = [[find_fold(l) for l in lineset[i]] for i in xrange(4)]
-        valid = [i for i in xrange(4) if any(f >= 0 for f in foldset[i])]
+        lineset = [get_lines(m, i) for i in range(4)]
+        foldset = [[find_fold(l) for l in lineset[i]] for i in range(4)]
+        valid = [i for i in range(4) if any(f >= 0 for f in foldset[i])]
 
         # TODO: Update random tile generation to account for new pick-three implementation
         maxval = m.max()
         if maxval >= 7 and random.random() < 1/24.:
             if maxval <= 9:
-                tileset = range(4, maxval-2)
+                tileset = list(range(4, maxval-2))
             else:
-                top = random.choice(xrange(6, maxval-2))
-                tileset = range(top-2, top+1)
+                top = random.choice(range(6, maxval-2))
+                tileset = list(range(top-2, top+1))
```

Comment:
Although these are generally small ranges, I would still prefer if you did something like “if PY2: range = xrange” to avoid unwanted behavior differences and potential performance differences.